app/services/anonymous_shadow_creator.rb

```diff
@@ -76,4 +76,11 @@ def create_shadow!
       shadow
     end
   end
+
+  def resolve_username
+    username = I18n.t("anonymous").downcase
+    username = "anonymous" until UserNameSuggester.sanitize_username(username).present?
```

Comment:
Huh! I've never seen `until` used that way. I had to think about this a couple of seconds...
I guess it works, it's just looks weird. :slightly_smiling_face:

I think the following should do the same and -- at least for me -- seems to be easier to understand. What do you think?

```suggestion
    username = UserNameSuggester.sanitize_username(username).presence || "anonymous"
```

Reply:
That was a silly typo, I meant `unless` not `until`. I've pushed a fix for that typo, @gschlager does the new version of the code make sense?